compiler/formatter/src/existing_parentheses.rs

```diff
@@ -0,0 +1,244 @@
+                    let new_closing_parenthesis = {
+                        let UnformattedCst { child, whitespace } =
+                            split_whitespace(closing_parenthesis);
+                        trailing_whitespace.push(whitespace);
+
+                        let mut whitespace = trailing_whitespace.remove(0);
+                        for trailing_whitespace in trailing_whitespace.drain(..) {
+                            trailing_whitespace
+                                .into_empty_and_move_comments_to(edits, &mut whitespace);
+                        }
+                        UnformattedCst { child, whitespace }
+                    };
+
+                    parentheses = match parentheses {
+                        ExistingParentheses::None { .. } => ExistingParentheses::Some {
+                            opening: new_opening_parenthesis,
+                            closing: new_closing_parenthesis,
+                        },
+                        ExistingParentheses::Some {
+                            opening: old_opening_parenthesis,
+                            closing: old_closing_parenthesis,
+                        } => {
+                            pub fn merge<'a>(
+                                edits: &mut TextEdits,
+                                mut left_parenthesis: UnformattedCst<'a>,
+                                right_parenthesis: UnformattedCst<'a>,
+                            ) -> UnformattedCst<'a> {
+                                if left_parenthesis.whitespace.has_comments() {
+                                    edits.delete(right_parenthesis.child.data.span.to_owned());
+                                    right_parenthesis
+                                        .whitespace
+                                        .into_empty_and_move_comments_to(
+                                            edits,
+                                            &mut left_parenthesis.whitespace,
+                                        );
+                                    left_parenthesis
+                                } else {
+                                    edits.delete(left_parenthesis.child.data.span.to_owned());
+                                    left_parenthesis.whitespace.into_empty_trailing(edits);
+                                    right_parenthesis
+                                }
+                            }
+                            ExistingParentheses::Some {
+                                opening: merge(
+                                    edits,
+                                    old_opening_parenthesis,
+                                    new_opening_parenthesis,
+                                ),
+                                closing: merge(
+                                    edits,
+                                    new_closing_parenthesis,
+                                    old_closing_parenthesis,
+                                ),
```

Comment:
Yes: The order is the same as they appear in the source code so that we prefer to keep the left-most parenthesis and move all comments there if necessary